src/web/modules/transactions/update/update.http.ts

```diff
@@ -0,0 +1,48 @@
+import { Request, Response } from 'express'
+import { parseString } from 'fast-csv'
+import moment from 'moment'
+import logger from '../../../../lib/logger'
+
+export async function fileUpload(req: Request, res: Response): Promise<void> {
+  res.render('transactions/update/upload.njk', {
+    csrf: req.csrfToken(),
+    messages: req.flash('info'),
+    errors: req.flash('error')
+  })
+}
+
+type TransactionRow = {
+  transaction_id: string;
+  event_date: string;
+}
+
+export async function update(req: Request, res: Response): Promise<void> {
+  let validationError = false
+  parseString<TransactionRow, TransactionRow>(req.file.buffer.toLocaleString(), { headers: true })
+    .validate((row, cb) => {
+      if (!row.transaction_id) {
+        return cb(null, false, 'transaction_id is missing')
+      }
+      if (row.event_date && !moment(row.event_date, moment.ISO_8601).isValid()) {
```

Comment:
Looks like a good pattern to me, we had also discussed adding default values for anything other than `transaction_id` (this would be `event_date` and `event_type`) but this functionality could be added in a future PR when it's easier to practically test end to end.

Reply:
Ah cool yeh, I wasn't entirely sure whether that was done on the toolbox side or not. Will add in a future PR.